getnacelle/nacelle-js
# chore(shopify-cart): replace estimatedCost with cost

## WHY are these changes introduced?

Addresses [ENG-6847](https://nacelle.atlassian.net/browse/ENG-6847) - replaces deprecated `estimatedCost` fields in queries with `cost`and updates types. 



## WHAT is this pull request doing?

1. Updates the `cart` fragment to use `cost` everywhere `estimatedCost` was used, and adds the additional fields available on `cost`.
2. Regenerates & updates the types to use `cost`. 
3. Updates the tests to fill in `cost` instead of `estimatedCost`. 





## How to Test

1. Pull this branch locally `mdarrik/shopify-cart/ENG-6847-replace-estimated-cost-with-cost` & navigate to `packages/shopify-cart`
2. Run `npm run test`. All tests should pass
3. Run `npm run build`. Build should succeed without issues.
4. Run `npm link` and then use/navigate to a project using `shopify-cart` and test it. Should get `cost` instead of `estimatedCost` in the lineItems response and the top level cart.

<details>
<summary>
Example Response from Prairie Winds Apparel
</summary>

```json
{
  "id": "gid://shopify/Cart/9a9b2cafc0269eeb25a9b2288507579e",
  "checkoutUrl": "https://pepper-wood-apparel.myshopify.com/cart/c/9a9b2cafc0269eeb25a9b2288507579e",
  "createdAt": "2022-08-26T18:40:20Z",
  "updatedAt": "2022-08-26T18:40:21Z",
  "buyerIdentity": {
    "countryCode": "US",
    "customer": null,
    "email": null,
    "phone": null
  },
  "lines": [
    {
      "id": "gid://shopify/CartLine/82a4ec09b10d9b452104d20dfb8eddab?cart=9a9b2cafc0269eeb25a9b2288507579e",
      "quantity": 2,
      "attributes": [],
      "cost": {
        "subtotalAmount": {
          "currencyCode": "USD",
          "amount": "280.0"
        },
        "totalAmount": {
          "currencyCode": "USD",
          "amount": "280.0"
        },
        "amountPerQuantity": {
          "currencyCode": "USD",
          "amount": "140.0"
        },
        "compareAtAmountPerQuantity": {
          "currencyCode": "USD",
          "amount": "160.0"
        }
      },
      "discountAllocations": [],
      "merchandise": {
        "id": "gid://shopify/ProductVariant/39717253283975",
        "availableForSale": true,
        "compareAtPriceV2": {
          "currencyCode": "USD",
          "amount": "160.0"
        },
        "priceV2": {
          "currencyCode": "USD",
          "amount": "140.0"
        },
        "requiresShipping": false,
        "title": "Default Title",
        "image": {
          "id": "gid://shopify/ProductImage/16284808151175",
          "url": "https://cdn.shopify.com/s/files/1/0344/4362/4583/products/photo-1485875437342-9b39470b3d95.jpg?v=1587622565",
          "altText": "Schuyler Glasses",
          "width": 1600,
          "height": 1064
        },
        "product": {
          "handle": "schuyler-glasses",
          "onlineStoreUrl": "https://pepper-wood-apparel.myshopify.com/products/schuyler-glasses",
          "tags": [
            "women"
          ],
          "title": "Schuyler Glasses - 3",
          "vendor": "Prairie Wind Apparel"
        },
        "selectedOptions": [
          {
            "name": "Title",
            "value": "Default Title"
          }
        ]
      }
    },
    {
      "id": "gid://shopify/CartLine/f9629fcbb724459e03f83375f9828321?cart=9a9b2cafc0269eeb25a9b2288507579e",
      "quantity": 2,
      "attributes": [],
      "cost": {
        "subtotalAmount": {
          "currencyCode": "USD",
          "amount": "210.0"
        },
        "totalAmount": {
          "currencyCode": "USD",
          "amount": "210.0"
        },
        "amountPerQuantity": {
          "currencyCode": "USD",
          "amount": "105.0"
        },
        "compareAtAmountPerQuantity": {
          "currencyCode": "USD",
          "amount": "135.0"
        }
      },
      "discountAllocations": [],
      "merchandise": {
        "id": "gid://shopify/ProductVariant/33894089064583",
        "availableForSale": true,
        "compareAtPriceV2": {
          "currencyCode": "USD",
          "amount": "135.0"
        },
        "priceV2": {
          "currencyCode": "USD",
          "amount": "105.0"
        },
        "requiresShipping": false,
        "title": "Default Title",
        "image": {
          "id": "gid://shopify/ProductImage/16284891512967",
          "url": "https://cdn.shopify.com/s/files/1/0344/4362/4583/products/pexels-photo-1536619.jpg?v=1587622825",
          "altText": "Lowry Denim Jacket",
          "width": 1600,
          "height": 1067
        },
        "product": {
          "handle": "lowry-denim-jacket",
          "onlineStoreUrl": "https://pepper-wood-apparel.myshopify.com/products/lowry-denim-jacket",
          "tags": [
            "women"
          ],
          "title": "Lowry Denim Jacket",
          "vendor": "Prairie Wind Apparel"
        },
        "selectedOptions": [
          {
            "name": "Title",
            "value": "Default Title"
          }
        ]
      }
    }
  ],
  "cost": {
    "checkoutChargeAmount": {
      "currencyCode": "USD",
      "amount": "490.0"
    },
    "subtotalAmount": {
      "currencyCode": "USD",
      "amount": "490.0"
    },
    "subtotalAmountEstimated": true,
    "totalAmount": {
      "currencyCode": "USD",
      "amount": "490.0"
    },
    "totalAmountEstimated": true,
    "totalDutyAmount": null,
    "totalDutyAmountEstimated": true,
    "totalTaxAmount": null,
    "totalTaxAmountEstimated": true
  },
  "attributes": [],
  "discountCodes": []
}
```

</details>

## Testing checklist

- [x] Could follow own testing steps
